<launch>

  <arg name="rviz" default="0" />
  <arg name="load_test_course" default="0" />

  <group ns="/uav" if="$(arg load_test_course)">
      <rosparam command="load" file="$(find spline_planner)/config/test_course.yaml"/>
  </group>

  
  <!-- Gate pose publisher node (cheat gate pose node) -->
  <node pkg="cascaded_pid_control" type="cheat_gate_locations_node.py" name="gt_gates_publisher" output="screen" />

  <!-- Planner node -->
  <node pkg="spline_planner" type="planner_new.py" name="spline_planner" output="screen">
    <remap from="/spline_planner/odometry" to="/CheatOdometryNode/odometry"/>
    <remap from="/spline_planner/trajectory" to="/CascadedPidControl/trajectory" />
  </node>

  <!-- ego trajectory publisher -->
  <node pkg="drone_map_builder" type="gt_ego_position.py" name="gt_ego_position" output="screen">
  </node>

  <!-- rviz node -->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find drone_race_common)/configs/solution_flightgoggles.rviz" if="$(arg rviz)"/>

  <!-- trajectory visualization topic publisher -->
  <node pkg="spline_planner" type="trajectory_visualizer.py" name="traj_viz">
    <remap from="~trajectory" to="/CascadedPidControl/trajectory"/>
    <remap from="~odometry" to="/CheatOdometryNode/odometry" />
  </node>  

</launch>